src/services/completions.ts

```diff
@@ -2370,6 +2370,24 @@ namespace ts.Completions {
                     return isPropertyDeclaration(contextToken.parent);
             }
 
+            function isPreviousPropertyDeclarationTerminated(contextToken: Node, previousToken: Node) {
+                return getLinesBetweenPositions(sourceFile, contextToken.end, previousToken.end) > 0 ||
+                    contextToken.kind === SyntaxKind.SemicolonToken;
+            }
+
+            // If we are inside a class declaration and typing `constructor` after property declaration...
+            if (contextToken !== previousToken
+                && isClassLike(previousToken.parent.parent)
+                && isPropertyDeclaration(contextToken.parent)
+                // After considering different contextToken...
+                && (isIdentifier(contextToken) || isTypeKeyword(contextToken.kind))

```

Comment:
Is this condition necessary/correct? If the previous property declaration has a type annotation or an initializer, there are lots of ways it might be something other than an identifier or type keyword. But those cases seem to work already... are we able to just drop this line?

Reply:
Correct. This is already supported by some other code. Cases like
```ts
type SomeType = number
class C {
  blah: SomeType; /**/
}
```
are already working in 4.3.0-beta.